ILTransform/TestProject.cs

```diff
@@ -1164,54 +1182,71 @@ in projectList.Zip(projectDirs, extraRootNames))
 
                         project.NewAbsolutePath = newProjectPath;
                         Utils.FileMove(project.AbsolutePath, newProjectPath, overwrite: false);
+                        if (project.CompileFilesIncludeProjectName)
+                        {
+                            _projects_to_update_ilfiles.Add(project);
+                        }
                     }
                 }
             }
         }
 
-        // Side effects: renames <NewTestClassSourceFile>, sets NewTestClassSourceFile and CompileFiles[0]
+        // Update IL filenames for each compile il using project name (as "$(MSBuildProjectName)")
+        public void FixILFilesWithProjectNames()
+        {
+            foreach (TestProject testProject in _projects_to_update_ilfiles)
```

Comment:
```suggestion
            foreach (TestProject testProject in _projects.Where(p => (p.NewAbsolutePath != null) && p.CompileFilesIncludeProjectName))
```

Note that there are some projects under `src\tests\Loader\classloader\StaticVirtualMethods` that use `MSBuildThisFileName` that will also potentially need this treatment.  (and perhaps whatever else `MSBuildProjectName` is used for)

Reply:
Is MSBuildThisFileName the same as MSBuildProjectName  ?

Reply:
In these cases, I think they have the same value.  I assume that they don't when you have multiple msbuild files used for a project.